conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/SerDe.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.google.common.reflect.TypeToken;
+import com.palantir.conjure.java.lib.SafeLong;
+import com.palantir.ri.ResourceIdentifier;
+import com.palantir.tokens.auth.BearerToken;
+import io.undertow.server.HttpServerExchange;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+/** Serialization functionality consumed by generated code. */
+public interface SerDe {
+
+    /** Serialize a value to a provided exchange. */
+    void serialize(Object value, HttpServerExchange exchange) throws IOException;
+
+    /** Serialize a {@link BinaryResponseBody} to <pre>application/octet-stream</pre>. */
+    void serialize(BinaryResponseBody value, HttpServerExchange exchange) throws IOException;
+
+    /** Deserializes the request body into the requested type. */
+    <T> T deserialize(TypeToken<T> type, HttpServerExchange exchange) throws IOException;
+
+    /** Reads an {@link InputStream} from the {@link HttpServerExchange} request body. */
+    InputStream deserializeInputStream(HttpServerExchange exchange);
+
+    // query, path, and header parameter deserializers
+
+    BearerToken deserializeBearerToken(String in);
```

Comment:
probably worth at least a TODO to doc this now that it's public API

Reply:
Good idea, will do